p2p/src/simulated/network.rs

```diff
@@ -90,10 +88,17 @@ impl<E: Spawner + Rng + Clock> Network<E> {
                 received_messages.clone(),
             );
         }
+
         let (oracle_sender, oracle_receiver) = mpsc::unbounded();
+        let next_addr = SocketAddr::new(
```

Comment:
nit: Recommend adding a comment as to why you did it this way (I think that way is fair but should record context)